src/Bridges/Constraint/norm_spec_nuc_to_psd.jl

```diff
@@ -0,0 +1,199 @@
+function concrete_bridge_type(::Type{<:NormSpectralBridge{T}}, G::Type{<:MOI.AbstractVectorFunction}, ::Type{MOI.NormSpectralCone}) where T
+    F = MOIU.promote_operation(vcat, T, MOIU.scalar_type(G), T)
+    return NormSpectralBridge{T, F, G}
+end
+
+# Attributes, Bridge acting as a model
+MOI.get(bridge::NormSpectralBridge{T, F, G}, ::MOI.NumberOfConstraints{F, MOI.PositiveSemidefiniteConeTriangle}) where {T, F, G} = 1
+MOI.get(bridge::NormSpectralBridge{T, F, G}, ::MOI.ListOfConstraintIndices{F, MOI.PositiveSemidefiniteConeTriangle}) where {T, F, G} = [bridge.psd_index]
+
+# References
+MOI.delete(model::MOI.ModelLike, bridge::NormSpectralBridge) = MOI.delete(model, bridge.psd_index)
+
+# Attributes, Bridge acting as a constraint
+function MOI.get(model::MOI.ModelLike, ::MOI.ConstraintFunction, bridge::NormSpectralBridge{T, F, G}) where {T, F, G}
+    psd_func = MOIU.eachscalar(MOI.get(model, MOI.ConstraintFunction(), bridge.psd_index))
+    t = psd_func[1]
+    side_dim = bridge.row_dim + bridge.column_dim
+    X = psd_func[[trimap(i, j) for j in 1:bridge.column_dim for i in (bridge.column_dim + 1):side_dim]]
+    return MOIU.convert_approx(G, MOIU.operate(vcat, T, t, X))
+end
+MOI.get(model::MOI.ModelLike, ::MOI.ConstraintSet, bridge::NormSpectralBridge) = MOI.NormSpectralCone(bridge.row_dim, bridge.column_dim)
+MOI.supports(::MOI.ModelLike, ::MOI.ConstraintPrimalStart, ::Type{<:NormSpectralBridge}) = true
+function MOI.get(model::MOI.ModelLike, attr::Union{MOI.ConstraintPrimal, MOI.ConstraintPrimalStart}, bridge::NormSpectralBridge)
+    primal = MOI.get(model, attr, bridge.psd_index)
+    t = primal[1]
+    side_dim = bridge.row_dim + bridge.column_dim
+    X = primal[[trimap(i, j) for j in 1:bridge.column_dim for i in (bridge.column_dim + 1):side_dim]]
+    return vcat(t, X)
+end
+function MOI.set(model::MOI.ModelLike, ::MOI.ConstraintPrimalStart, bridge::NormSpectralBridge{T}, value) where T
+    column_dim = bridge.column_dim
+    side_dim = bridge.row_dim + column_dim
+    primal = zeros(T, div(side_dim * (side_dim + 1), 2))
+    for i in 1:side_dim
+        primal[trimap(i, i)] = value[1]
+    end
+    X_idx = 2
+    for j in 1:column_dim, i in (column_dim + 1):side_dim
+        primal[trimap(i, j)] = value[X_idx]
+        X_idx += 1
+    end
+    MOI.set(model, MOI.ConstraintPrimalStart(), bridge.psd_index, primal)
+    return
+end
+# Given [U X'; X V] is dual on PSD constraint, the dual on NormSpectralCone
+# constraint is (tr(U) + tr(V), 2X) in NormNuclearCone.
+function MOI.get(model::MOI.ModelLike, ::MOI.ConstraintDual, bridge::NormSpectralBridge)
+    dual = MOI.get(model, MOI.ConstraintDual(), bridge.psd_index)
+    column_dim = bridge.column_dim
+    side_dim = bridge.row_dim + column_dim
+    t = sum(dual[trimap(i, i)] for i in 1:side_dim)
+    X = 2 * dual[[trimap(i, j) for j in 1:column_dim for i in (column_dim + 1):side_dim]]
+    return vcat(t, X)
+end
+
```

Comment:
For ConstraintDualStart, I would set all `trimap(i, i)` to the dual of `t` divided by `2side_dim`. It might not be the only possibility but it works.

Reply:
OK should I make another PR for that?

Reply:
I'm wrong, that won't make the matrix PSD.
In fact, it's easier to discuss the ConstraintPrimalStart of NormNuclearBridge.
So suppose you have X and t set, we can assume that t >= sum sigma_i(X)
[A X'; X B] >= 0 is equivalent to A >= X' inv(B) X. If X = U S V', then, it's equivalent to
V' * A * V <= S * U' * inv(B) * U * S
then, by choosing A = V * S * V' and B = U * S * U', you get
S <= S * inv(S) * S which is satisfied.
As the trace must sum up to t, you set A = V * S * V' * (t / tr(S)) and B = U * S * U' * (t / tr(S).

Reply:
if that works for constraint primal start of NormNuclear then should it also work for constraint dual start of NormSpectral?

Reply:
Yes, it was just easier to reason in the primal than in the dual ^^

Reply:
OK rather than having you explain the details to me, it's probably easier if you implement those new start functions